src/devices_models/devices/AC_branches.jl

```diff
@@ -289,15 +317,30 @@ function add_constraints!(
     radial_network_reduction = get_radial_network_reduction(network_model)
     radial_branches_names = PNM.get_radial_branches(radial_network_reduction)
 
-    for r in rating_data
-        if r[1] ∈ radial_branches_names
+    use_slacks = get_use_slacks(device_model)
+    if use_slacks
+        slack_ub = get_variable(container, FlowActivePowerSlackUpperBound(), B)
+    end
+
+    for (branch_name, branch_rate) in rating_data
+        if branch_name ∈ radial_branches_names
             continue
         end
-        for t in time_steps
-            constraint[r[1], t] = JuMP.@constraint(
-                get_jump_model(container),
-                var1[r[1], t]^2 + var2[r[1], t]^2 <= r[2]^2
-            )
+        if use_slacks
+            for t in time_steps
```

Comment:
This feels like it should be two functions.

Reply:
I would have to duplicate a lot more code or write functions that accept a lot of arguments